src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -528,5 +549,71 @@ public void Dispose()
 
         private string NormalizePath(string path) =>
             path == null ? null : Path.GetFullPath(path.Replace('\\', Path.DirectorySeparatorChar).Replace('/', Path.DirectorySeparatorChar));
+
+        public static string GetLatestVSMSBuildPath()
+        {
+            var programFilesX86 = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86);
+            var programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+            var windows = Environment.GetFolderPath(Environment.SpecialFolder.Windows);
+
+            var editions = new string[] { "Enterprise", "Professional", "Community" };
+            var targets = new string[] { "Microsoft.CSharp.targets", "Microsoft.CSharp.CurrentVersion.targets", "Microsoft.CSharp.CrossTargeting.targets" };
+
+            foreach (string editionType in editions)
+            {
+                var vsPath2022 = Path.Combine(programFiles, @"Microsoft Visual Studio\2022\" + editionType + @"\MSBuild\Current\Bin\");
+                var vsPath2019 = Path.Combine(programFilesX86, @"Microsoft Visual Studio\2019\" + editionType + @"\MSBuild\Current\Bin\");
+                var vsPath2017 = Path.Combine(programFilesX86, @"Microsoft Visual Studio\2017\" + editionType + @"\MSBuild\15.0\Bin\");
+
+                var msBuild2022 = Path.Combine(vsPath2022, "MSBuild.exe");
+                var msBuild2019 = Path.Combine(vsPath2019, "MSBuild.exe");
+                var msBuild2017 = Path.Combine(vsPath2017, "MSBuild.exe");
```

Comment:
Not major one but some people might still use vs2015, good to add the path for vs2015 as well

Reply:
msbuild 14 is the path for VS 2015.